app/models/exchange.rb

```diff
@@ -95,4 +97,10 @@ def role
   def participant
     incoming? ? sender : receiver
   end
+
+  def touch_receiver
+    return unless receiver&.persisted?
+
+    receiver.touch_later
+  end
```

Comment:
I haven't manually checked it but it seems that Rails deals with `touch_later` transparently for us when the `touch: true` association option is specified. If that's right, we would be implementing Rails here. Have you checked that?

It was https://github.com/rails/rails/commit/e7c48db509f1b712c2a72a3ff7f27e3e8b7b4868 that lead me to think so. The PR that introduced the feature provides extra context as well :point_right: https://github.com/rails/rails/pull/19324

Reply:
I think `touch_later` is introduced in Rails 5.0, but that transparent batching and related bits aren't actually present in our current version of `ActiveRecord`, we'll get more of those benefits when we upgrade to 5.2.

Reply:
Have you tried putting a `byebug` and digging deeper into Rails to see that? I'm in a rush if not I'd do it myself

Reply:
I had a good look at the source code in our current gems and compared it with some of those sexy Rails PRs, and we don't have it yet. They start implementing it in 5.0, but the **good stuff** doesn't really land until 5.1/5.2.

Reply:
We just need to wait a little bit longer. Almost there!